docs/bills-matching.md

```diff
@@ -0,0 +1,139 @@
+
+The algorithm is held by the [`Linker`](), and more precisely its `linkBillsToOperations` method. This method takes 3 parameters:
+
+* An array of `io.cozy.bills` objects
+* An array of `io.cozy.bank.operations` objects
+* An options object
+
+The options object can contain the following properties:
+
+* `minAmountDelta`: defaults to `0.001`
+* `maxAmountDelta`: defaults to `0.001`
+* `pastWindow`: defaults to 15 (days)
+* `futureWindow`: defaults to 29 (days)
+
+These options will determine the default date and amount windows in which a
+transaction should be to match with a bill.
+
+## General flow
+
+When called, this method will do the following (each part will be explained in
+details after):
+
+* Filter bills that are declared as third party payer: these bills have no matching transation since it has been paid by somebody else
+* Then for each bill:
+  * Try to find a matching debit transaction (amount < 0)
+  * Try to find a matching credit transaction (amount > 0) if the bill is a refund
+  * Try to combine bills that didn't match anything to see if they match something when combined
+  * Update transactions that matched in database
+
+## Finding debit and credit operations for a bill
+
+The algorithm is globally the same to find a debit or a credit operation for a
+bill, but there are some differences at some steps.
+
+### Find neighboring transactions
+
+First, we look for transactions that are « neighbors » of the bill. It means
+the ones that fits in the amount and date windows of the bill.
+
+The date window is `[bill date - dateLowerDelta; bill date + dateUpperDelta]`. Where:
+
+* `bill date` is `bill.date` if we are looking for a credit; `bill.originalDate` with a fallback on `bill.date` if we are looking for a debit
+* `dateLowerDelta` is `bill.matchingCriterias.dateLowerDelta` if it exists, `options.pastWindow` otherwise
+* `dateUpperDelta` is `bill.matchingCriterias.dateUpperDelta` if it exists, `options.futureWindow` otherwise
+
+The amount window is `[bill amount - amountLowerDelta; bill amount + amountUpperDelta]`. Where:
+
+* `bill amount` is `bill.groupAmount` with a fallback on `bill.amount` if we are looking for a credit; `-bill.originalAmount` with a fallback on `-bill.amount` if we are looking for a debit
+* `amountLowerDelta` is `bill.matchingCriterias.amountLowerDelta` if it exists, `options.minAmoutDelta` otherwise
+* `amountUpperDelta` is `bill.matchingCriterias.amountUpperDelta` if it exists, `options.maxAmoutDelta` otherwise
+
+All transactions that matches these two criterias are kept.
+
+### Apply filters to neighboring transactions
+
+There are other filters a transaction must pass to match with a bill.
+
+#### Category
+
+Health bills are a little special because they can only match with health
+transactions. So if the bill is an health bill, we keep only health
+transactions. Otherwise, we keep all but health transactions.
+
+You can pass a `allowUncategorized` option to the algorithm, in which case
```

Comment:
```suggestion
`allowUncategorized` can be passed option to the algorithm, in which case
```

Before, the doc is not written as a guide but as a technical documentation, this is why I think we should avoid the "You"